Automattic/jetpack
# WAF: Add Brute force protection module

#### Description
Moves the Jetpack Protect (Brute force protection) module from the Jetpack plugin to the WAF package.

## Proposed changes:

* Moves primary Protect code out of the Jetpack module
* Adds the module code to the WAF package
* Updates all external Jetpack code reliant on the module to use the WAF package version
* Removes any unused external code
* Adds namespacing for the primary classes

### Other information:


## Jetpack product discussion


- 1201069996155224-as-1203716242675353
- 1203716242675355-as-1203716242675347

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Checkout this branch
* Start up your Jurassic Tube
* Enable Jetpack and approve a user connection
* Proceed to your Jetpack Settings and verify that the Brute force module was enabled by default
* Proceed to your admin login page in an incognito browser window
* Continuously fail the login and ensure you are eventually met with a math authentication prompt
* Continue to fail the math authentication login until you are effectively blocked by Brute force
* Submit the account recovery email form and login into the admin screen
* In a separate incognito browser window, proceed to your admin login screen and ensure you are still blocked
* Proceed to add your IP address to the Brute force protection whitelist in your Jetpack settings
* Once again return to the login screen in an incognito window and verify that you are now able to access the login form and that failed attempts are no longer met with a math authentication prompt or a block
* Verify that the Jetpack dashboard Brute force widget displays details about the blocked attempts
* Remove your IP from the whitelist, disabled the module, and ensure that the login screen is not blocked
* Ensure that there are regressions in functionality introduced
